### PR TITLE
Validate Tweet Message Length

### PR DESCRIPTION
**Pull request by donaldali**
Fix for issue #173 
Add validation setting a maximum length of 140 for the tweet message on
lesson create and update. Include test and modify controller to handle
a lesson that fails validation.

Check the final tweet message composed in `tweet` method of
`lib/lesson_tweeter.rb`, ensuring that a tweet is not sent if the final
composed tweet message exceeds 140 characters.

### CODE REVIEW

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -87,7 +90,10 @@ def update
         format.html { redirect_to @lesson, notice: 'Lesson was successfully updated.' }
         format.json { head :no_content }
       else
-        format.html { render action: "edit" }
+        format.html do
+          @venues = current_school.venues
+          render action: "edit"
+        end
```
Comment (gkop) [inline app/controllers/lessons_controller.rb]: Is there a better way to factor the view code so we don't need to set `@venues =` all over?

**Comment by gkop** [comment]
+1

File: app/controllers/lessons_controller.rb (hunk 1 above)
Comment (donaldali) [inline app/controllers/lessons_controller.rb]: @venues is used once in the form for lessons, so it should be possible to add a line in the form partial to replace all the `@venues =` used in the controller. Testing it...

Comment (gkop) [inline app/controllers/lessons_controller.rb]: Thanks. I know it's tough to decide when to refactor in a messy code base, maybe this is a good opportunity,

**Pull request merged**